### ISSUE TITLE
Database Adapter - EMC2 and EcoSense

### ISSUE THREAD

**Issue by Ludee** [ReeemProject/reeem_db#7]
- [x] [**adatpter coming soon**]
- [x] [**reeem_io.py**](https://github.com/ReeemProject/reeem_db/blob/master/database_adapter/reeem_io.py)

**Comment by doroschmid** [comment]
I tested the adapters and they work just fine (did not test EMC2 output data, since it is such a big file and it takes some while to upload it to the database, but it started, so I guess it all works).

How is it now with the schemas? As far as I understood, all model data are now stored under model_draft and we need to add another column to include the best fitting schema (I guess for better sorting)?

Then I have a question regarding the nid column: Is it meant to be a continues id per file (unique identifier for each row), or is it meant to be the same number for each row in one file/table (more or less numbering the files/tables within one file)?

In the next week, I would adapt the structure then and add all the missing metadata.

**Comment by Ludee** [comment]
Thanks for the test, happy that it is working!

The **nid** should be used to identify the **indicator**. So each indicator should have the same nid on each sheet.

**Comment by Ludee** [comment]
There is a model factsheet:
http://oep.iks.cs.ovgu.de/factsheets/models/145/
please review and complete infos @doroschmid

**Comment by Ludee** [comment]
I had to modify the CSV-files for Ecosense:
* Add a unique id (dfid)
* Fill column region_2 with "none" for the empty rows

**Comment by doroschmid** [comment]
I just updated the metadata (hopefully for the last time) and inlcuded the sources with links and a better description of the models. Sorry for the many commits and I unfortunately forgot to reference the issue.

**Comment by doroschmid** [comment]
Just tried to upload the new dataset to the database.
@Ludee / @4lm is it possible that the dfid column does not exist in the database?

**Comment by Ludee** [comment]
@doroschmid Yes, the column "dfid" was missing. I fixed it and restarted the upload.

**Comment by 4lm** [comment]
Hi @doroschmid,

I'm checking the metadata for all models right now and I realized, that in the metadata string of the input metadata, the values for the temporal section are missing: https://github.com/ReeemProject/reeem_db/blob/master/database_setup/reeem_db_setup_ecosense_eva.sql#L56-L60

Can you help?

**Comment by doroschmid** [comment]
Hej @4lm 

I added 'NA' for the temporal section because for the input data it is impossible to determine one reference data and also not a start and end date because of the complexity of the data and the two models used. Some data are held, for example, constant over time (e.g. background disease rates) and are always taken from the latest available year, other data are extrapolated (e.g. population data) but are originally based in again a different reference year.

So I hope 'NA' in this case is okay, to just have something in there? Or should it be 'None' then?

**Comment by 4lm** [comment]
Hi @doroschmid,

thanks for the info and the commit!

Yes, it should be 'none' (all lower caps). Would you like to change all the 'NA' to 'none' or shall I do it?

**Comment by 4lm** [comment]
Hi @Ludee, the output section in the setup script and the concerning table in the SQL database has an attribute source. Is this correct? I thought only input data has sources, because the source of the output is the result of the processed model (EcoSenseEVA). BTW, source is attributed as "EcoSense". Maybe a special case? Please, shad some light onto this - thanks!

**Comment by doroschmid** [comment]
Hej @4lm,
I'm just gonna jump in on this because it is a special case and I'm not sure if @Ludee still remembers the reason for the source column. The data in this table is coming from two models (EcoSense and EVA), so the source column is there two indicate which model is responsible for which data. For the first dataset, only EcoSense is used but there should be a DataV2 that also has 'EVA' in the source column for some rows....

**Comment by 4lm** [comment]
Hi @doroschmid,

thanks for the quick clarification why the "source" column in the output!

Indeed, "EcoSense" and "EVA" is present in the "source" column in the database as values!

**Comment by 4lm** [comment]
IMPORTANT: Request for OEP-Publishing

Hi again @doroschmid,

you licensed reeem_ecosenseeva_output as CC-BY-4.0 and therefore ready for publishing on the Open Energy Platform (OEP). We now pre-staged the data on OEP and are ready to hit the button :) Nevertheless we want to ask for your final OK!

If you want to have a look at the data, you find it in the REEEM DB in model_draft.reeem_ecosenseeva_output

Thanks in advance!

**Comment by doroschmid** [comment]
Hej,

I've been using the data from the database the last month so I think it should be good to go.

So hit the button ;-)

Will you add a link here to the published data? That would really be great!
